### DISCUSSION TITLE
IWXXM Support to Hazardous Weather Information Services (HWIS)

### DISCUSSION THREAD

**Discussion by blchoy** [team-sx/iwxxm-sx#4] [Ideas]
In the ICAO METP WG-MRAD WS-HWIS web meeting on 6 Sep 2023, Clemens Weidemann introduced the latest concept on creating the data set for HWIS.  The following two slides are most interesting:

![Slide_8](https://github.com/team-sx/iwxxm-sx/assets/30064374/91066094-0c7d-4e1b-92c7-1eb50f7b124f)

![Slide_9](https://github.com/team-sx/iwxxm-sx/assets/30064374/2099fa1d-649b-4a59-b956-3cd1cdcc0e6e)

He further indicated that _"...HWIS integrators do not gain data superiority as tey collect national and local inputs - their assumed role is closer to an ROC than a VAAC."_

From IWXXM perspective, the WxObject approach should already be able to handle the above arrangements.  In particular:

1. Each contributor provide an iwxxm:MeteorologicalFeatureCollection which contains multiple iwxxm:MeteorologicalFeature (also know as the WxObject) indicating the "polygons" (or more specifically 2D horizontal surface or 2.5D volume) of hazardous weather.  This approach has also been adopted in WAFS SIGWX F/C and the upcoming QVACI.

<p align="center"><img src="https://github.com/team-sx/iwxxm-sx/assets/30064374/bdc5947c-48bf-47a1-ba88-680c4df6b713"></p>

2. Note that each WxObject also allows inclusion of the identify of the originator.

3. The integrator, after receiving the iwxxm:MeteorologicalFeatureCollection prepared by the contributors, will extract the WxObjects.

4. As mentioned by Clemens, the integrator will "fill in the regional gaps" by creating new WxObjects from WAFS data.  The integrator may also want to concatenate "polygons" for cross-border weather phenomenon.  In the latter case, while the originator of the WxObject of the concatenated "polygon" should be the integrator, back references to the source WxObjects can be included (remember each of the WxObject has a unique identifier defined by gml:identifier) for reference as well as a trail for cost recovery.

5. The integrator will finally prepare an iwxxm:MeteorologicalFeatureCollection of all its WxObjects for distribution, as well as making the WxObjects available on web services.

Glad to see if Clemens or his team has any further views of this initial idea.

**Comment by cwdmnn** [comment]
Hi Choy, really appreciate your approach here! It seems like the current IWXXM capabilities enable our theoretical concept (in principle). This is very good new in my point of view. Kind regards, Clemens

**Comment by blchoy** [comment]
Thanks Clemens.  We are about 2/3 of the way to create a workable IWXXM schema for HWIS.  To move on, we need to:

1. Define the representation for contributor and integrator identities (e.g. just VHHH or more sophisticated representation like allowing the inclusion of its responsible "polygon").

2. Confirm the currently defined geometries are adequate to describe the "polygons":

![Basic Types](https://github.com/team-sx/iwxxm-sx/assets/30064374/537a4910-77c0-40da-9b71-a90c7a48f1a1)

3. Define the representation for HWIS weather phenomena (we can make reference to what has been done for [WAFS SIGWX F/C](https://schemas.wmo.int/iwxxm/2023-1/html/index.htm?goto=8:148) and QVACI).

4. Identify the activities involved, information services to be provided and associated use cases.  This is necessary as we need this information to fine tune the structure of a WxObject (e.g. if an integrator creates consolidated "polygon" across border, do we need to make back references to the original WxObjects?).

You can see that Items (1)-(3) are pretty straight forward.  Item (4) is going to be matured during the course of HWIS concept development.

I hope this gives you a better idea of our expectation in developing IWXXM for HWIS.

**Comment by blchoy** [comment]
I have presented a paper IP3103 at ICAO METP WG-MIE/11 meeting during 4-7 Dec 2023.  It contains more detailed design of the IWXXM information model to support HWIS operation:

[MIE11-IP3103_NewIWXXMDesign.pdf](https://github.com/team-sx/iwxxm-sx/files/13593379/MIE11-IP3103_NewIWXXMDesign.pdf)

It also exposed the question on the necessity for contributors "sending" hazardous weather objects in a collection to an integrator where they subsequently "send" a collection of consolidated objects to downstream users.  If request-reply protocol is being deployed there is no need to create the collections at all, and it may not even required a physical integrator; an "integrator API" should be able to make request to upstream contributors and collate the objects.

It is interesting to note that use cases drives the design of an information service which in term drives the design of the payload.

**Comment by blchoy** [comment]
Closed because it has been dormant for quite some time.
